api/youtube_downloader.py

```
@@ -313,33 +367,48 @@ def _populate_video_info(self) -> bool:
         logger.debug(f"Extracted metadata: Title='{self.video_metadata['title']}', Channel='{self.video_metadata['channel']}', Views={self.video_metadata.get('view_count', 'Unknown')}, Date='{self.video_metadata.get('publish_date', 'Unknown')}'")
 
         # Extract subtitle tracks
+        logger.debug("Starting subtitle tracks extraction from player response")
         captions_data = self.player_response.get("captions")
+        logger.debug(f"Captions data present: {captions_data is not None}")
+        
```

Pylint (3.3.7)

[convention] Trailing whitespace

(C0303)